Simplified/NYPLBookState.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+
+let DownloadingKey = "downloading"
+let DownloadFailedKey = "download-failed"
+let DownloadNeededKey = "download-needed"
+let DownloadSuccessfulKey = "download-successful"
+let UnregisteredKey = "unregistered"
+let HoldingKey = "holding"
+let UsedKey = "used"
+let UnsupportedKey = "unsupported"
+
+@objc public enum NYPLBookState : Int {
+  case Unregistered = 0
+  case DownloadNeeded = 1
+  case Downloading
+  case DownloadFailed
+  case DownloadSuccessful
+  case Holding
+  case Used
+  case Unsupported
+    
+  init?(_ stringValue: String) {
+    switch stringValue {
+      case DownloadingKey:
+        self = .Downloading
+      case DownloadFailedKey:
+        self = .DownloadFailed
+      case DownloadNeededKey:
+        self = .DownloadNeeded
+      case DownloadSuccessfulKey:
+        self = .DownloadSuccessful
+      case UnregisteredKey:
+        self = .Unregistered
+      case HoldingKey:
+        self = .Holding
+      case UsedKey:
+        self = .Used
+      case UnsupportedKey:
+        self = .Unsupported
+      default:
+        return nil
+    }
+  }
+    
+  func stringValue() -> String {
+    switch self {
+      case .Downloading:
+        return DownloadingKey;
+      case .DownloadFailed:
+        return DownloadFailedKey;
+      case .DownloadNeeded:
+        return DownloadNeededKey;
+      case .DownloadSuccessful:
+        return DownloadSuccessfulKey;
+      case .Unregistered:
+        return UnregisteredKey;
+      case .Holding:
+        return HoldingKey;
+      case .Used:
+        return UsedKey;
+      case .Unsupported:
+        return UnsupportedKey;
+    }
+  }
+}
+
+// For Objective-C, since Obj-C enum is not allowed to have methods
+@objcMembers class NYPLBookStateHelper : NSObject {
+  static func getString(from state: NYPLBookState) -> String {
```

Comment:
this name is not very swift-y / objc-y: perhaps `stringValue(from:)` would be better. And I would add a `@objc(stringValueFromBookState:)` above the declaration for a more common objc name.

Reply:
@ettore is that some sort of decorator that can explicitly set the name used in ObjC?

Reply:
yep that's it... see for example NYPLBackgroundExecutor::setUpWorkItem(wrapping:). Very useful to keep objc code non-swifty. It's basically the longer form of the usual `@objc func ...`.

I also tend to avoid using @objcMembers because of that, and to avoid exposing too much to objc: I find it pushes me to write more swift rather than going the easy way of adding to existing objc sources.

Reply:
Updated the method name